src/shogun/clustering/KMeansLloydImpl.cpp

```diff
@@ -17,127 +18,140 @@ using namespace shogun;
 
 namespace shogun
 {
-void CKMeansLloydImpl::Lloyd_KMeans(int32_t k, CDistance* distance, int32_t max_iter, SGMatrix<float64_t> mus,
-		SGVector<int32_t> ClList, SGVector<float64_t> weights_set, bool fixed_centers)
+void CKMeansLloydImpl::Lloyd_KMeans(int32_t k, CDistance* distance, int32_t max_iter, SGMatrix<float64_t> mus, 
+		bool fixed_centers)
 {
 	CDenseFeatures<float64_t>* lhs=
 		CDenseFeatures<float64_t>::obtain_from_generic(distance->get_lhs());
-	int32_t XSize=lhs->get_num_vectors();
+	
+	int32_t lhs_size=lhs->get_num_vectors();
 	int32_t dimensions=lhs->get_num_features();
 
 	CDenseFeatures<float64_t>* rhs_mus=new CDenseFeatures<float64_t>(0);
 	CFeatures* rhs_cache=distance->replace_rhs(rhs_mus);
 
-	SGVector<float64_t> dists=SGVector<float64_t>(k*XSize);
-	dists.zero();
+	SGVector<int32_t> cluster_assignments=SGVector<int32_t>(lhs_size);
+	cluster_assignments.zero();
+
+	/* Weights : Number of points in each cluster */
+	SGVector<float64_t> weights_set=SGVector<float64_t>(k);
+	weights_set.zero();
+	/* Initially set all weights for zeroth cluster, Changes in assignement step */
+	weights_set[0]=lhs_size;
+
+	distance->precompute_lhs_squared_norms();
 
 	int32_t changed=1;
-	int32_t iter=0;
+	int32_t iter;
 	int32_t vlen=0;
 	bool vfree=false;
 	float64_t* vec=NULL;
 
-	while (changed && (iter<max_iter))
+	for(iter=0; iter<max_iter; iter++)
 	{
-		iter++;
 		if (iter==max_iter-1)
-			SG_SWARNING("kmeans clustering changed throughout %d iterations stopping...\n", max_iter-1)
+			SG_SWARNING("kmeans clustering has reached maximum number of ( %d ) iterations \n", iter)
 
-		if (iter%1000 == 0)
-			SG_SINFO("Iteration[%d/%d]: Assignment of %i patterns changed.\n", iter, max_iter, changed)
 		changed=0;
+		rhs_mus->copy_feature_matrix(mus);
 
+		distance->precompute_rhs_squared_norms();
+		
+		/* Assigment step : Assign each point to nearest cluster */
+		for (int32_t i=0; i<lhs_size; i++)
+		{ 
+			const int32_t cluster_assignments_i=cluster_assignments[i];
+			int32_t min_cl, j;
+			float64_t min_dist, dist;
+			
+			min_cl=0;
+		   	min_dist=distance->distance(i,0);
+			for (j=1; j<k; j++)
+			{
+				dist=distance->distance(i,j);
+				if (dist<min_dist)
```

Comment:
just realized while doing this , need the if block for the cluster assignment anyways ...